From 6b0987ae149f912be748a7ee8a63aeec5fabb156 Mon Sep 17 00:00:00 2001
Message-ID: <6b0987ae149f912be748a7ee8a63aeec5fabb156.1719691458.git.sharlatanus@gmail.com>
In-Reply-To: <cover.1719691458.git.sharlatanus@gmail.com>
References: <cover.1719691458.git.sharlatanus@gmail.com>
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Tue, 18 Jun 2024 18:30:08 +0100
Subject: [PATCH 16/26] gnu: Add go-github-com-neurosnap-sentences.

* gnu/packages/golang-xyz.scm (go-github-com-neurosnap-sentences, go-sentences): New variables.

Change-Id: I56c9e6c26567cfb9a6a21f8e637ddd1465d3bb58
---
 gnu/packages/golang-xyz.scm | 39 +++++++++++++++++++++++++++++++++++++
 1 file changed, 39 insertions(+)

diff --git a/gnu/packages/golang-xyz.scm b/gnu/packages/golang-xyz.scm
index 944d5f93f8..af7860781d 100644
--- a/gnu/packages/golang-xyz.scm
+++ b/gnu/packages/golang-xyz.scm
@@ -3438,6 +3438,31 @@ (define-public go-github-com-nbrownus-go-metrics-prometheus
 registry.")
     (license license:asl2.0)))
 
+(define-public go-github-com-neurosnap-sentences
+  (package
+    (name "go-github-com-neurosnap-sentences")
+    (version "1.1.2")
+    (source
+     (origin
+       (method git-fetch)
+       (uri (git-reference
+             (url "https://github.com/neurosnap/sentences")
+             (commit (string-append "v" version))))
+       (file-name (git-file-name name version))
+       (sha256
+        (base32 "0qkq635x54mqzydxmifh2l0kicacgqcbkw4vli1cnwwcs0x902f2"))))
+    (build-system go-build-system)
+    (arguments
+     (list
+      #:go go-1.21
+      #:import-path "github.com/neurosnap/sentences"))
+    (home-page "https://github.com/neurosnap/sentences")
+    (synopsis "Multilingual command line sentence tokenizer in Golang")
+    (description
+     "This package provides functionality of converting a blob of text into a
+list of sentences.")
+    (license license:expat)))
+
 (define-public go-github-com-nsqio-go-diskqueue
   (package
     (name "go-github-com-nsqio-go-diskqueue")
@@ -4785,6 +4810,20 @@ (define-public go-pixelmatch
      "This package provides a CLI build from the
 go-github-com-orisano-pixelmatch source.")))
 
+(define-public go-sentences
+  (package
+    (inherit go-github-com-neurosnap-sentences)
+    (name "go-sentences")
+    (arguments
+     (list
+      #:import-path "github.com/neurosnap/sentences/cmd/sentences"
+      #:unpack-path "github.com/neurosnap/sentences"
+      #:install-source? #f))
+    (description
+     (string-append (package-description go-github-com-neurosnap-sentences)
+                    "  This package provides an command line interface (CLI)
+tool."))))
+
 (define-public go-tengo
   (package
     (inherit go-github-com-d5-tengo-v2)
-- 
2.41.0

